android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -358,6 +367,67 @@ public void captureScreenshot(Promise promise) {
         promise.resolve(screenshotsArray);
     }
 
+    public void fetchViewHierarchy(Promise promise) {
+        final @Nullable Activity activity = getCurrentActivity();
+        final @Nullable View decorView = getCurrentDecorViewForViewHierarchy(activity);
+
+        if (decorView == null) {
+            promise.resolve(null);
+            return;
+        }
+
+        final @NotNull ViewHierarchy viewHierarchy = ViewHierarchyEventProcessor.snapshotViewHierarchy(decorView);
+
+        byte[] bytes;
+        try {
+            try (final @NotNull ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                 final @NotNull BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(stream, UTF_8));
+                 final @NotNull JsonObjectWriter jsonWriter = new JsonObjectWriter(writer, 100)) {
+
+                viewHierarchy.serialize(jsonWriter, logger);
+                jsonWriter.flush();
+                bytes = stream.toByteArray();
+            }
+        } catch (Throwable t) {
+            logger.log(SentryLevel.ERROR, "Could not serialize ViewHierarchy", t);
+            promise.resolve(null);
+            return;
+        }
+
+        if (bytes.length < 1) {
+            logger.log(SentryLevel.ERROR, "Got empty bytes array ViewHierarchy");
+            promise.resolve(null);
+            return;
+        }
+
+        final WritableNativeArray data = new WritableNativeArray();
+        for (final byte b : bytes) {
+            data.pushInt(b);
+        }
+        promise.resolve(data);
+    }
+
+    private @Nullable View getCurrentDecorViewForViewHierarchy(final @Nullable Activity activity) {
```

Comment:
M: This code already lives in the `ViewHierarchyEventProcessor` Android, maybe the processor exposes this function as static and RN can reuse it, right now it's duplicating code which increases bundle size.

Reply:
I've updated the view hierarchy event processor so all the checks are done in it.
https://github.com/getsentry/sentry-java/pull/2492